res/controllers/Ion-Discover-DJ-Pro-scripts.js

```diff
@@ -0,0 +1,262 @@
+// Ion Discover DJ pro Mapping Script Functions
+// Joachim
+var IonDiscoverDjPro = function() {};
+
+IonDiscoverDjPro.init = function(id) {	// called when the MIDI device is opened & set up
+    IonDiscoverDjPro.id = id;	// Store the ID of this device for later use
+
+    IonDiscoverDjPro.directoryMode = false;
+
+    IonDiscoverDjPro.scratchMode = [false, false];
+    IonDiscoverDjPro.scratchTimer = [-1, -1];
+
+    IonDiscoverDjPro.manualLooping = [false, false];
+
+    IonDiscoverDjPro.leds = [
+        // Common
+        { },
+        // Deck 1
+        {"rate": 0x70, "scratchMode": 0x48, "keylock": 0x51, "manualLoop": 0x61, "loopIn": 0x53, "loopOut": 0x54, "reLoop": 0x55},
+        // Deck 2
+        {"rate": 0x71, "scratchMode": 0x50, "keylock": 0x52, "manualLoop": 0x62, "loopIn": 0x56, "loopOut": 0x57, "reLoop": 0x58}
+    ];
+    IonDiscoverDjPro.setLED(IonDiscoverDjPro.leds[0]["file"], true);
```

Comment:
Isn't that value undefined?

Reply:
On second thought.... I think it would be good to add some dummy indexes to all array, so that there is no need to use the index "deck-1" in the code .This way we keep the same meaning of the deck index as used with the engine functions and readability of the code will improve (I think).